src/main/java/core/basesyntax/strategy/impl/BalanceOperation.java

```diff
@@ -0,0 +1,11 @@
+package core.basesyntax.strategy.impl;
+
+import core.basesyntax.strategy.OperationHandler;
+import java.util.Map;
+
+public class BalanceOperation implements OperationHandler {
+    @Override
+    public void apply(String fruit, int quantity, Map<String, Integer> storage) {
+        storage.put(fruit, quantity);
+    }

```

Comment:
Make sure to validate the inputs such as `fruit` and `quantity` before using them. It's important to check for null values or negative quantities to ensure data integrity.